File: Codes/2788-split-strings-by-separator.py
'''
Author: LetMeFly
Date: 2024-01-20 23:24:06
LastEditors: LetMeFly
LastEditTime: 2024-01-20 23:25:26
'''
from typing import List

class Solution:
    def splitWordsBySeparator(self, words: List[str], separator: str) -> List[str]:
        ans = []
        for word in words:
            splited = word.split(separator)
            for this in splited:
                if this:
                    ans.append(this)
        return ans

File: Codes/2788-split-strings-by-separator.cpp
/*
 * @Author: LetMeFly
 * @Date: 2024-01-20 23:18:28
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-01-20 23:22:03
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    vector<string> splitWordsBySeparator(vector<string>& words, char separator) {
        vector<string> ans;
        for (string& word : words) {
            int last = -1;
            for (int i = 0; i <= word.size(); i++) {
                if (i == word.size() || word[i] == separator) {
                    if (i - last > 1) {
                        ans.push_back(word.substr(last + 1, i - last - 1));
                    }
                    last = i;
                }
            }
        }
        return ans;
    }
};